internal/pkg/pkg.go

```diff
@@ -77,26 +92,177 @@ func New(path string) (*Pkg, error) {
 
 // Kptfile returns the Kptfile meta resource by lazy loading it from the filesytem.
 // A nil value represents an implicit package.
-func (p *Pkg) Kptfile() *kptfilev1alpha2.KptFile {
-	if !p.kptfileLoaded {
-		// TODO
-		// p.kptfile = ...
-		p.kptfileLoaded = true
+func (p *Pkg) Kptfile() (*kptfilev1alpha2.KptFile, error) {
+	if p.kptfile == nil {
+		kf, err := p.readKptfile()
+		if err != nil {
+			return nil, err
+		}
+		p.kptfile = kf
 	}
-	return p.kptfile
+	return p.kptfile, nil
 }
 
-// Pipeline returns the Pipeline meta resource by lazy loading it from the filesystem.
-func (p *Pkg) Pipeline() *kptfilev1alpha2.Pipeline {
-	if !p.pipelineLoaded {
-		// TODO
-		// p.pipeline = ...
-		p.pipelineLoaded = true
+// readKptfile reads the KptFile in the given pkg.
+// TODO(droot): This method exists for current version of Kptfile.
+// Need to reconcile with the team how we want to handle multiple versions
+// of Kptfile in code. One option is to follow Kubernetes approach to
+// have an internal version of Kptfile that all the code uses. In that case,
+// we will have to implement pieces for IO/Conversion with right interfaces.
+func (p *Pkg) readKptfile() (*kptfilev1alpha2.KptFile, error) {
+	kf := &kptfilev1alpha2.KptFile{}
+
+	f, err := os.Open(path.Join(p.UniquePath.String(), kptfilev1alpha2.KptFileName))
```

Comment:
Do you actually need `UniquePath.String()`? I find think of String() to be mostly relevant for printing purposes not for computation. Why not just `string(p.UniquePath)`.

Reply:
Changed this one to `string(p.UniquePath)`.